File: arm.64/frame/electron/src/bind.js
import { boot } from './boot.js';

export function bind(libDir) {
  const lib = boot(libDir);
  if (!lib) return null;
  return {
    lib,
    Attach: lib.func('bool Attach(void*)'),
    Detach: lib.func('bool Detach()'),
    Invoke: lib.func('int Invoke(const char*, const char*, const char*)'),
    Report: lib.func('bool Report(_Out_ Info* out)'),
  };
}

File: arm.64/apple/tablet/src/bind.swift
import Foundation

func bind(_ path: String) -> Abi? {
    guard let h = dlopen(path, RTLD_NOW) else { return nil }
    guard let a = dlsym(h, "attach"), let d = dlsym(h, "detach"),
          let i = dlsym(h, "invoke"), let r = dlsym(h, "report") else {
        dlclose(h)
        return nil
    }
    return Abi(handle: h,
               attach: unsafeBitCast(a, to: AttachFn.self),
               detach: unsafeBitCast(d, to: DetachFn.self),
               invoke: unsafeBitCast(i, to: InvokeFn.self),
               report: unsafeBitCast(r, to: ReportFn.self))
}
